torch/_dynamo/variables/lists.py

```diff
@@ -100,6 +101,49 @@ def call_method(
 
         return super(BaseListVariable, self).call_method(tx, name, args, kwargs)
 
+    def next_variables(self):
+        return unimplemented(f"{type(self)}.next_variables()")
+
+    @staticmethod
+    def generic_list_compare(left, tx, op, right, **options):
```

Comment:
The way I implemented this was by writing a generic list comparison (I think tuple vs. list comparison ops all do the same type of reduction?). The bits that I made different are:
- Each VariableTracker gets its own type check (tuple must be compared with tuple, list with list, but torch.Size vs. tuple comparison is ok)
- I implemented `curr, rest = next_variables()` on a few BaseListVariableTracker subclasses to do the reduction, which is container-dependent.

Reply:
Changed impl